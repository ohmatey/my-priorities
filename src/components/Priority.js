import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { compose, withHandlers } from 'recompose'

import { addPriorityById } from '../actions/'

export const Priority = ({ id, name, hours, addPriorityById }) => (
  <div className="card">
    <div className="card-content">
      <p className="title">{name}</p>
      <p className="subtitle">{hours} hrs</p>
    </div>
    <footer className="card-footer">
      <p className="card-footer-item" style={{ cursor: 'pointer' }} onClick={addPriorityById}>
        <span>
          Add Priority
        </span>
      </p>
    </footer>
  </div>
)

Priority.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  hours: PropTypes.number,
  addPriorityById: PropTypes.func
}

const enhance = compose(
  connect(),
  withHandlers({
    addPriorityById: ({ dispatch, id }) => e => dispatch(addPriorityById(id))
  })
)

export default enhance(Priority)
