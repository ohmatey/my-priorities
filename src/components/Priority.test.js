import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

import { Priority } from './Priority'

describe('PrioritiesList component', () => {
  it('matches snapshot', () => {
    const tree = shallowToJson(
      <Priority />
    )
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with props', () => {
    const tree = shallowToJson(
      <Priority id={0} name={'Work'} hours={2} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('handles on click', () => {
    const mock = jest.fn()
    const component = shallow(
      <Priority id={0} name={'Work'} hours={2} addPriorityById={mock} />
    )
    component.find('.card-footer-item').simulate('click')
    expect(mock).toBeCalled()
  })
})
