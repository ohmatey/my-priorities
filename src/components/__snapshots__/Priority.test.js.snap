// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrioritiesList component matches snapshot 1`] = `undefined`;

exports[`PrioritiesList component matches snapshot with props 1`] = `undefined`;
